src/app/classes/Processor/TemplateProcessor.php

```diff
@@ -34,85 +35,91 @@ class TemplateProcessor
     private $config;
 
     /**
+     * @var PickerFactory
+     */
+    private $pickerFactory;
+
+    /**
      * TemplateProcessor constructor.
      * @param Api $api
      * @param CollectionInterface $config
+     * @param PickerFactory $pickerFactory
      */
-    public function __construct(Api $api, CollectionInterface $config)
+    public function __construct(
+        Api $api,
+        CollectionInterface $config,
+        PickerFactory $pickerFactory
+    )
     {
         $this->config = $config;
         $this->api = $api;
+        $this->pickerFactory = $pickerFactory;
     }
 
     /**
      * @param array $templatesContent
      *
      * @return array
      */
-    protected function generateFlatData($templatesContent)
+    protected function generateFlatData(array $templatesContent)
     {
         $entitiesList = array();
 
-        if (is_array($templatesContent)) {
-            foreach ($templatesContent as $template) {
-                $label = $template['label'];
-                $templateDefinition = $template['definition'];
+        foreach ($templatesContent as $template) {
+            //TODO: add validation or better schema validation
+            $label = $template['label'];
+            $templateDefinition = $template['definition'];
 
-                $templateDefinition = array_merge($templateDefinition,
-                    $this->config->get('data_generator_export_configuration'));
+            $templateDefinition = array_merge(
+                $templateDefinition,
+                $this->config->get('data_generator_export_configuration')
+            );
 
-                $definition = json_encode($templateDefinition);
+            $definition = json_encode($templateDefinition);
 
-                $entitiesList[$label] = json_decode($this->api->apiCall($definition), true);
-            }
+            $apiResult = $this->api->apiCall($definition);
+            $entitiesList[$label] = json_decode($apiResult, true);
         }
 
         return $entitiesList;
     }
 
     /**
-     * @param string $template
+     * @param array $template
+     *
      * @return array|null
+     * @throws \ErrorException
      */
-    public function processTemplate($template)
+    public function processTemplate(array $template)
     {
-        foreach ($template as $type => $settings) {
```

Comment:
I added the foreach loop intentionally, because before (and now) the order of processing the template elements (templates, relations, postprocessors) is hard-coded. By looping with foreach you could place the postprocessors *before* the relations wihch is necessary e.g. for the magento template.
We should probably discuss a new approach.